llvm/include/llvm/ADT/GenericConvergenceVerifier.h

```diff
@@ -63,6 +63,8 @@ template <typename ContextT> class GenericConvergenceVerifier {
   // and not the token values.
   DenseMap<const InstructionT *, const InstructionT *> Tokens;
 
+  bool SeenFirstConvOp = false;
```

Comment:
The value will not be reset across basic block, right?

Reply:
It gets reset at the start of every block in the [visit method](https://github.com/ssahasra/llvm-project/blob/fa6dd7a24af2b02f236ec3b980d9407e86c2c4aa/llvm/include/llvm/IR/GenericConvergenceVerifierImpl.h#L76).

Reply:
I think not? The visit() method you pointed out is only called in Verifier::visitCallBase(). So it cannot reset the flag in a new block.

Reply:
Ouch! You are right. I was paying too much attention on the MachineConvergenceVerifier (not checked in yet), and didn't see how the ConvergenceVerifier traverses the function. I have a fix in my branch, will push it out real soo now. Thanks for catching it!